pkg/controllers/nsxserviceaccount/nsxserviceaccount_controller.go

```diff
@@ -29,6 +34,73 @@ import (
 	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/nsxserviceaccount"
 )
 
+const proxyLabelKey = "mgmt-proxy.antrea-nsx.vmware.com"
+
+var (
+	proxyServicePred = predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldObj := e.ObjectOld.(*corev1.Service)
+			newObj := e.ObjectNew.(*corev1.Service)
+
+			newLabels := e.ObjectNew.GetLabels()
+			oldLabels := e.ObjectOld.GetLabels()
+			_, oldObjHasLabel := oldLabels[proxyLabelKey]
+			_, newObjHasLabel := newLabels[proxyLabelKey]
+
+			// Reconcile all NSXSA if label "mgmt-proxy.antrea-nsx.vmware.com" added to or removed from existing Service
+			if oldObjHasLabel != newObjHasLabel {
+				return true
+			}
+
+			// Reconcile all NSXSA if existing Service with label "mgmt-proxy.antrea-nsx.vmware.com" is updated with Service LB VIP
+			if oldObjHasLabel && newObjHasLabel {
+				// Service type updated to LB
+				if oldObj.Spec.Type != corev1.ServiceTypeLoadBalancer && newObj.Spec.Type == corev1.ServiceTypeLoadBalancer {
```

Comment:
what if oldObj.Spec.Type != newObj.Spec.Type and both are not corev1.ServiceTypeLoadBalancer, will this be ignored, for example, service type has been changed from nodeport to clusterip

Reply:
Yes, it will be ignored.